cpp/src/parquet/file_writer.cc

```diff
@@ -288,45 +256,48 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
   }
 
   void InitColumns() {
-    for (int i = 0; i < num_columns(); i++) {
+    for (int i = 0; i < RowGroupSerializer::num_columns(); i++) {
```

Comment:
Is there a particular reason for this change?

Reply:
`num_columns()` is a virtual function ( I don't know the reason ). And `InitColumns` is called in ctor. So I think we'd better write it explicitly